libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/IndividualFeatureImportanceView/LargeIndividualFeatureImportanceView/LocalImportanceChart.tsx

```diff
@@ -0,0 +1,274 @@
+            <Stack.Item>
+              {this.props.isLocalExplanationsDataLoading ? (
+                <LoadingSpinner label={localization.Counterfactuals.loading} />
+              ) : (
+                <BasicHighChart
+                  configOverride={this.getLocalImportanceBarOptions()}
+                  id={"WhatIfFeatureImportanceBar"}
+                />
+              )}
+            </Stack.Item>
+          </Stack>
+        </Stack.Item>
+        <Stack.Item>
+          <Stack horizontal={false} tokens={{ childrenGap: "m1" }}>
+            <Stack.Item className={classNames.absoluteValueToggle}>
+              <Toggle
+                label={localization.Interpret.GlobalTab.absoluteValues}
+                inlineLabel
+                checked={this.state.sortAbsolute}
+                onChange={this.toggleSortAbsolute}
+                disabled={this.props.isLocalExplanationsDataLoading}
+              />
+            </Stack.Item>
+            {IsClassifier(this.props.modelType) && (
+              <Stack.Item>
+                <ClassImportanceWeights
+                  onWeightChange={this.onWeightChange}
+                  selectedWeightVector={this.props.selectedWeightVector}
+                  weightOptions={this.props.weightOptions}
+                  weightLabels={this.props.weightLabels}
+                  disabled={this.props.isLocalExplanationsDataLoading}
+                />
+              </Stack.Item>
+            )}
+          </Stack>
+        </Stack.Item>
+      </Stack>
+    );
+  }
+
+  private setTopK = (newValue: number): void => {
+    this.setState({ topK: newValue });
+  };
+
+  private generateSortedData(): void {
+    const sortedData: Array<{ [key: string]: number[] | number | undefined }> =
+      [];
+    if (IsClassifier(this.props.modelType)) {
+      sortedData.push({
+        [this.props.weightOptions[0]]: this.getAbsoluteValues(
+          this.props.data?.precomputedExplanations?.localFeatureImportance
+            .scores[0][0]
+        )
+      });
+      sortedData.push(...this.addScores());
+    } else {
+      sortedData.push({
+        [regressionKeyValue]:
+          this.props.data?.precomputedExplanations?.localFeatureImportance
+            .scores[0]
+      });
+    }
+
+    this.setState({ sortedData });
+  }
+
+  private getAbsoluteValues(values?: number[]): number[] | undefined {
+    if (!values) {
+      return values;
+    }
+    const sortedScores = values.map((score) => Math.abs(score));
+    return sortedScores;
+  }
+
+  private addScores(): Array<{ [key: string]: number[] }> {
+    const scores: Array<{ [key: string]: number[] }> = new Array(
+      this.props.data?.precomputedExplanations?.localFeatureImportance.scores.length
+    );
+    this.props.data?.precomputedExplanations?.localFeatureImportance.scores.forEach(
+      (score, index) => {
+        scores.push({ [this.props.weightOptions[index + 1]]: score[0] });
+      }
+    );
+    return scores.filter((s) => s);
```

Comment:
what does this filter do? filter undefined s? can we check if score[0] is defined at line 210?

Reply:
Thanks Tong, I used scores.filter((s) => s) as push was adding items on top of the length of the array and hence was creating empty items. I updated to use [index] and hence filter is not needed now.